src/Language/Haskell/LSP/Core.hs

```diff
@@ -179,24 +173,22 @@ instance E.Exception ProgressCancelledException
 -- @since 0.11.0.0
 data ProgressCancellable = Cancellable | NotCancellable
 
-data SomeServerMessageWithResponse where
-  SomeServerMessageWithResponse
-    :: (J.ToJSON (J.ServerMessage m))
-    => J.SServerMethod m
-    -> J.ServerMessage m
-    -> ServerResponseHandler m
-    -> SomeServerMessageWithResponse
-
-type ServerMessageFunc = SomeServerMessageWithResponse -> IO ()
-
 -- | Returned to the server on startup, providing ways to interact with the client.
 data LspFuncs c =
   LspFuncs
     { clientCapabilities           :: !C.ClientCapabilities
     , config                       :: !(IO (Maybe c))
       -- ^ Derived from the DidChangeConfigurationNotification message via a
       -- server-provided function.
-    , sendFunc                     :: !ServerMessageFunc
+    , sendReq                      :: !(forall m.
```

Comment:
I think we could break out these types as synonyms, so that documenting them would be easier.